rotkehlchen/chain/ethereum/modules/balancer/balancer.py

```diff
@@ -0,0 +1,710 @@
+                **param_values,
+                'offset': param_values['offset'] + GRAPH_QUERY_LIMIT,  # type: ignore
+            }
+
+        protocol_balance = ProtocolBalance(
+            address_balances=dict(address_balances),
+            known_tokens=known_tokens,
+            unknown_tokens=unknown_tokens,
+        )
+        return protocol_balance
+
+    def _get_known_token_prices(self, known_tokens: Set[EthereumToken]) -> TokenPrices:
+        """Get the USD price of the known tokens"""
+        token_prices: TokenPrices = {}
+        for token in known_tokens:
+            usd_price = Inquirer().find_usd_price(token)
+            if usd_price == Price(ZERO):
+                self.msg_aggregator.add_error(
+                    f"Failed to request the USD price of {token.identifier}. "
+                    f"Balances of the balancer pools that have this token won't be accurate.",
+                )
+                continue
+
+            token_prices[token.ethereum_address] = usd_price
+        return token_prices
+
+    def _get_trades(
+            self,
+            addresses: List[ChecksumEthAddress],
+            from_timestamp: Timestamp,
+            to_timestamp: Timestamp,
+            only_cache: bool,
+    ) -> AddressTrades:
+        """Given a list of addresses returns all their trades
+        (1 AMMTrade has N AMMSwap) within the given time range. Requests all the
+        swaps of the new addresses via subgraph. Requests all the new swaps
+        (since last used query range) of the existing addresses via subgraph.
+
+        May raise RemoteError
+        """
+        address_swaps: AddressSwaps = {}
+        new_addresses: List[ChecksumEthAddress] = []
+        existing_addresses: List[ChecksumEthAddress] = []
+        min_end_ts: Timestamp = to_timestamp
+
+        if only_cache:
+            return self._fetch_trades_from_db(
+                addresses=addresses,
+                from_timestamp=from_timestamp,
+                to_timestamp=to_timestamp,
+            )
+
+        # Get addresses' last used query range for Balancer trades
+        for address in addresses:
+            entry_name = f'{BALANCER_TRADES_PREFIX}_{address}'
+            trades_range = self.database.get_used_query_range(name=entry_name)
+
+            if not trades_range:
+                new_addresses.append(address)
+            else:
+                existing_addresses.append(address)
+                min_end_ts = min(min_end_ts, trades_range[1])
+
+        # Request new addresses' trades
+        if new_addresses:
+            start_ts = Timestamp(0)
+            new_address_swaps = self._get_swaps_graph(
+                addresses=new_addresses,
+                start_ts=start_ts,
+                end_ts=to_timestamp,
+            )
+            address_swaps.update(new_address_swaps)
+            self._update_used_query_range(
+                addresses=new_addresses,
+                prefix='balancer_trades',
```

Comment:
It is a literal, and it happens the same with AMMTradeLocation. I'll see if it can be done better

Reply:
This can be done very easilly by declaring `BALANCER_TRADES_PREFIX` to have a literal type of 'balancer_trades'`. https://mypy.readthedocs.io/en/stable/literal_types.html#declaring-literal-variables

I know .. kind of stupid but this is how constants and literals work now. I hope things will improve as mypy and python improve.